# Magma bug patch PHP022 (generated by gen-patch, magma-agent)
# target:   php
# pinned:   e49be5f8e2903dbefa262847f1494c8e4c86004e
# upstream: 9ed82b1f7b 2020-03-16 "Fixed bug #79282"
# cve:      CVE-2020-7064
# class:    uninit_read
# site:     ext/exif/exif.c:exif_process_TIFF_in_JPEG
# canary:   length < 2
# harness:  exif (existing)
# Lines above the first 'diff --git' are ignored by patch(1) and git-apply.
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -3734,10 +3734,15 @@ static void exif_process_TIFF_in_JPEG(image_info_type *ImageInfo, char *CharBuf,
 	unsigned exif_value_2a, offset_of_ifd;
 	exif_offset_info info;
 
+	#ifdef MAGMA_ENABLE_CANARIES
+		MAGMA_LOG("%MAGMA_BUG%", length < 2);
+	#endif
+	#ifdef MAGMA_ENABLE_FIXES
 	if (length < 2) {
 		exif_error_docref(NULL EXIFERR_CC, ImageInfo, E_WARNING, "Missing TIFF alignment marker");
 		return;
 	}
+	#endif
 
 	/* set the thumbnail stuff to nothing so we can test to see if they get set up */
 	if (memcmp(CharBuf, "II", 2) == 0) {
@@ -3889,8 +3894,12 @@ static bool exif_scan_JPEG_header(image_info_type *ImageInfo)
 			return false;
 		}
 
+	#ifdef MAGMA_ENABLE_FIXES
 		sn = exif_file_sections_add(ImageInfo, marker, itemlen, NULL);
-		Data = ImageInfo->file.list[sn].data;
+	#else
+		sn = exif_file_sections_add(ImageInfo, marker, itemlen+1, NULL);
+	#endif
+	Data = ImageInfo->file.list[sn].data;
 
 		/* Store first two pre-read bytes. */
 		Data[0] = (uchar)lh;
